sip/sip-006-runtime-cost-assessment.md

```diff
@@ -746,3 +754,130 @@ types is defined as:
+
+If it is decided that a published function should be a part of the Clarity
+language, it can then be re-implemented as a native function in Rust and
+included in a future release of Clarity. Nodes running this upgraded VM
+would instead use this new native implementation of the function when they
+encounter Clarity code that references it via `contract-call?`.
+
+The new native function is likely faster and more efficient than the
+prior Clarity implementation, so a new cost-assessment function may need
+to be chosen for its evaluation. Until a new cost-assessment function is
+agreed upon, the cost of the Clarity implementation will continue to be used.
+
+## Voting on the Cost of Clarity Functions
+
+New and more accurate cost-assessment functions can be agreed upon as
+follows:
+
+1. A user formulates a cost-assessment function and publish it in a
+Clarity contract as a `read-only` function.
+2. The user proposes the cost-assessment function by calling the
+`propose-cost` function in the **Clarity Cost Voting Contract**.
+3. Voting on the proposed function ensues via the voting functions in
+the **Clarity Cost Voting Contract**, and the function is either
+selected as a replacement for the existing cost-assessment function,
+or ignored.
+
+### Publishing a Cost-Assessment Function
+
+Cost-assessment functions to be included in proposals can be published
+in Clarity contracts. They must be written precisely as follows:
+
+1. The function should be `read-only`.
+2. The function should return a tuple with the keys `runtime`, `write_length`,
+`write_count`, `read_count`, and `read_length` (the execution cost measurement
+[categories](#measurements-for-execution-cost)).
+3. The values of the returned tuple should be Clarity expressions representing
+the asymptotic cost functions.
+3. The expression's constants and variables should be listed as arguments
+to the Clarity function.
+    * The arguments should be written in alphabetical order, and variables
+    should come before constants.
+
+Here is an example of a cost-assessment function where the runtime
+is set to `a * log(n) + b`, and all of the other parameters are constant.
+
+```Lisp
+(define-read-only (runtime_cost (n int) (a int) (b int))
+  {
+     runtime: (+ (* a (log n)) b)
+     write_length: a,
+     write_count: a,
+     read_count: a,
+     read_length: a,
+  })
+```
+
+### Making a Cost-Assessment Function Proposal
+
+Stacks holders can propose cost-assessment functions by calling the
+`propose-cost` function in the **Clarity Cost Voting Contract**. It
+can be called like so:
+
+```Lisp
+(contract-call?
+   .propose-cost
+   qualified-function-name
+   cost-function-address
+   cost-function-name)
+
+```
+
+This function will return a response containing the details of the proposal,
+and a proposal ID, if successful.
+
+Once submitted, a proposal is valid until either a supermajority miner vote
+is achieved, or until it expires, 14 days from when it was submitted.
```

Comment:
How is the 14 days measured?  I would suggest replacing "14 days" with "2016 Bitcoin blocks," since that's a measure of time all Stacks nodes on all forks can see.

Reply:
Will do! Also, do you have an opinion on the length? 14 was kind of an arbitrary pick... I figured it should be some number of days, probably less than a month, but I wasn't really sure how to decide.